models/user.go

```diff
@@ -19,7 +19,10 @@ func CreateUser(u *User) (id int64, err error) {
 
 func (u *User) IsExistingUser() bool {
 	o := orm.NewOrm()
-	err := o.Read(u, "Email")
+	var user User
+
+	err := o.QueryTable(User{}).Filter("email__iexact", u.Email).One(&user)
```

Comment:
A different alternative is to use `citext` for the column data type: https://hashrocket.com/blog/posts/working-with-email-addresses-in-postgresql#another-approach. It's usually what I recommend for case insensitive pieces of data such as email.

Reply:
I am using Beego's [automatic table generation](https://beego.me/docs/mvc/model/orm.md#registering-model) using `RunSyncdb`. With this I don't need to create any migration to create/update tables. Beego does it automatically by mapping model fields  with database type when the app runs(it does in runtime)

But [Beego](https://beego.me/docs/mvc/model/models.md) does not support `string` to `citext` mapping. For `string` model field type, it only supports `varchar, char, text, json, jsonb`

To use `citext` I need to switch back to manually migration(with `bee` CLI), where I can define `citext` column in raw sql. Also drop automatic table generation code and generate all tables from migrations.

What do you think?
![Screen Shot 2021-03-24 at 10 31 05 AM](https://user-images.githubusercontent.com/6076762/112250927-e037de80-8c8c-11eb-914a-dbfe34fc8494.png)

Reply:
Noted. So a custom implementation does makes sense.